common/fee/utils.go

```diff
@@ -0,0 +1,23 @@
+package fee
+
+import (
+	"math/big"
+
+	bigmath "github.com/smartcontractkit/chainlink/v2/core/utils/big_math"
+
+	"github.com/shopspring/decimal"
+)
+
+func GetMaxFeePrice(userSpecifiedMax, maxFee *big.Int) *big.Int {
+	return bigmath.Min(userSpecifiedMax, maxFee)
+}
+
+func CapFeePrice(calculatedFeePrice, userSpecifiedMax, maxFeePriceWei *big.Int, feeLimit uint32, multiplier float32) (maxFeePrice *big.Int, chainSpecificFeeLimit uint32) {
```

Comment:
So we are assuming that in a generalized form, these are the types for gas units:
feePrice: big.Int
feeLimit: uint32

These seem to be ok for EVM.
But in future, we may have to tweak those for other non-evms.

I don't want to block the PR for this, but just keep in mind, these types may have to converted to generics at some point in future.

Reply:
yes @prashantkumar1982, we are assuming that in a generalized form, these are the types for gas units:
feePrice: big.Int
feeLimit: uint32

And it would have to be changed for non evm types - i.e use generics. 

Right now generics can make things harder to read than normal typed code and it enforces less strictly which is why i did not want to go with generics.

Will put that in the comments in the future! 

Thanks for pointing that out :)

Reply:
Have we considered other names for `feeLimit`? This sounds to me like something that limits a fee, both expressed in the same units, but that would actually be `feeLimit * maxFeePrice`.

Reply:
Agreed that the name might not be the most intuitive, but it was a 1:1 mapping from `gasLimit` to `feeLimit`, and i thought it might too confusing if i changed it to something else?